rayon-core/src/configuration.rs

```diff
@@ -6,48 +6,29 @@ use std::fmt;
 use std::sync::Arc;
 use registry;
 
-/// Custom error type for the rayon thread pool configuration.
+/// Error if the gloal thread pool is initialized multiple times.
 #[derive(Debug,PartialEq)]
-pub enum InitError {
-    /// Error if number of threads is set to zero.
-    NumberOfThreadsZero,
+pub struct GlobalPoolAlreadyInitialized;
```

Comment:
I'm happy with making this a struct, but I think it should still be an opaque `InitError`, with at least some private placeholder contents so it can't be created externally.  Or maybe even use `Box<Error>` where we need it, and let the actual error type be fully private.  We want rayon-core to be compatible ~forever.